### PR TITLE
feat(ExtensionFilter): change the filter in extension

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
## Description

create a new Filter for Extension that follow the same Structure as Resource
- the extension Filter doesn't use CostumFilter
- the autoComplet/authSuggestion are not developed yet.

Filter Bar
image

Criterias Selector :
image

image

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

see if the ExtensionFilter folow the same Structure as Resource
and see if all the functions are working.
see if applyFilter atom is receving the chosed Criterias

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/Extensions/Filter/Criterias/models.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  value?: CriteriaValue;
+}
+
+const criteriaValueNameById = {
+  INSTALLED: labelInstalled,
+  MODULE: labelModule,
+  OUTDATED: labelOutdated,
+  UPDETED: labelUpdated,
+  WIDGET: labelWidget,
+  not_installed: labelNotInstalled,
```
Comment (Thebarda) [inline www/front_src/src/Extensions/Filter/Criterias/models.tsx:28]:
```suggestion
  NOT_INSTALLED: labelNotInstalled,
```

File: www/front_src/src/Extensions/Filter/Criterias/models.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+const installedId = 'INSTALLED';
+const installed = { id: installedId, name: criteriaValueNameById[installedId] };
+
+const notInstalledId = 'not_installed';
```
Comment (Thebarda) [inline www/front_src/src/Extensions/Filter/Criterias/models.tsx:34]:
```suggestion
const notInstalledId = 'NOT_INSTALLED';
```

File: www/front_src/src/Extensions/Filter/Criterias/models.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+export interface Criteria {
+  name: string;
+  value?: CriteriaValue;
+}
+
+const criteriaValueNameById = {
+  INSTALLED: labelInstalled,
+  MODULE: labelModule,
+  OUTDATED: labelOutdated,
+  UPDETED: labelUpdated,
```
Comment (Thebarda) [inline www/front_src/src/Extensions/Filter/Criterias/models.tsx:26]:
```suggestion
  UPDATED: labelUpdated,
```

File: www/front_src/src/Extensions/Filter/Criterias/Criteria.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        field="name"
+        // getEndpoint={getEndpoint}
+        isOptionEqualToValue={isOptionEqualToValue}
+        value={value}
+        onChange={(_, updatedValue): void => {
+          changeCriteria(updatedValue);
+        }}
+      />
+    );
+  }
```
Comment (Thebarda) [inline www/front_src/src/Extensions/Filter/Criterias/Criteria.tsx:78]:
```suggestion
```
I think we can remove this Autocomplete

File: www/front_src/src/Extensions/Filter/Criterias/searchQueryLanguage/models.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+const statusNameToQueryLanguageName = selectableStatuses
+  .map(prop('id'))
+  .reduce((previous, current) => {
+    return { ...previous, [current]: toLower(current) };
+  }, {});
+
+export const criteriaNameToQueryLanguageName = {
+  ...statusNameToQueryLanguageName,
+  resource_type: 'type', // ????
+  unhandled_problems: 'unhandled', /// ????
```
Comment (Thebarda) [inline www/front_src/src/Extensions/Filter/Criterias/searchQueryLanguage/models.tsx:33]:
```suggestion
```

File: www/front_src/src/Extensions/Filter/index.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        <div className={classes.container}>
+          <React.Suspense
+            fallback={
+              <LoadingSkeleton height={24} variant="circular" width={24} />
+            }
+          >
+            <Criterias />
+          </React.Suspense>
+          <SearchField
+            // fullWidth // ???
```
Comment (Thebarda) [inline www/front_src/src/Extensions/Filter/index.tsx:114]: Is the search bar is pretty complex due to the query language, we need to expand the search bar to the full width possible

File: www/front_src/src/Extensions/Filter/index.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    // const arrowDownKeyPressed = event.key === 'ArrowDown';
+    // const arrowUpKeyPressed = event.key === 'ArrowUp';
+    const arrowLeftKeyPressed = event.key === 'ArrowLeft';
+    const arrowRightKeyPressed = event.key === 'ArrowRight';
+
+    if (arrowLeftKeyPressed || arrowRightKeyPressed) {
+      updateCursorPosition();
+    }
+
+    // The rest need to add AutoSuggestion feature
```
Comment (Thebarda) [inline www/front_src/src/Extensions/Filter/index.tsx:97]: I assume you need to implement the rest of function here :)

File: www/front_src/src/Extensions/Filter/Criterias/models.tsx (hunk 1 above)
Comment (Yassir-BenBOUBKER) [reply www/front_src/src/Extensions/Filter/Criterias/models.tsx:28]: eslint demande That i follow one of this snake_case, camelCase, PascalCase

File: www/front_src/src/Extensions/Filter/index.tsx (hunk 7 above)
Comment (Yassir-BenBOUBKER) [reply www/front_src/src/Extensions/Filter/index.tsx:97]: thanks, for now i dont have data to complete autoComplete and Auto Suggestion

File: www/front_src/src/Extensions/Filter/Criterias/models.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  value?: CriteriaValue;
+}
+
+const criteriaValueNameById = {
+  INSTALLED: labelInstalled,
+  MODULE: labelModule,
+  OUTDATED: labelOutdated,
+  UPDATED: labelUpdated,
+  WIDGET: labelWidget,
+  not_installed: labelNotInstalled, // eslint doesn't accept NOT_INSTALLED
```
Comment (bdauria) [inline www/front_src/src/Extensions/Filter/Criterias/models.tsx:28]: Then use NOTINSTALLED instead

File: www/front_src/src/Extensions/Filter/index.tsx [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+      <CloseIcon color="action" fontSize="small" />
+    </IconButton>
+  );
+};
+
+const Filter = (): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+
+  // const [isSearchFieldFocus, setIsSearchFieldFocused] = React.useState(false);
```
Comment (bdauria) [inline www/front_src/src/Extensions/Filter/index.tsx:57]: Remove comments

File: www/front_src/src/Extensions/translatedLabels.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+export const labelClear = 'Clear';
+export const labelSearch = 'Search';
+export const labelSearchOptions = 'Filter options';
+export const labelStatus = 'Status';
+export const labelType = 'Types';
+export const labelInstalled = 'Installed';
+export const labelNotInstalled = 'NotInstalled';
```
Comment (bdauria) [inline www/front_src/src/Extensions/translatedLabels.ts:7]:
```suggestion
export const labelNotInstalled = 'Not installed';
```

File: www/front_src/src/Extensions/Filter/Criterias/models.tsx (hunk 8 above)
Comment (jdelpierre) [reply www/front_src/src/Extensions/Filter/Criterias/models.tsx:28]: why your comment "eslint ..." ?

**Pull request closed without merging**